lib/kubeclient/common.rb

```diff
@@ -78,11 +92,47 @@ def handle_exception
       raise KubeException.new(e.http_code, err_message, e.response)
     end
 
+    def discover
+      @entities = {}
+      response = handle_exception do
+        rest_client.get(@headers)
+      end
```

Comment:
@moolitayer will a one-liner be nicer here?

```ruby
response = handle_exception { rest_client.get(@headers) }
```

Same for other similar instances in the rest of the PR.